helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -418,6 +418,8 @@ public ClusterConfig getClusterConfig() {
   public void setClusterConfig(ClusterConfig clusterConfig) {
     _clusterConfig = clusterConfig;
     refreshAbnormalStateResolverMap(_clusterConfig);
+    updateIdealRuleMap(_clusterConfig);
+    updateDisabledInstances(getInstanceConfigMap().values(), _clusterConfig);
```

Comment:
Then why this place we don't use `getClusterConfig()`. 

Either way it is find to me. But please do use one approach.

Reply:
I can change to "clusterConfig" instead of "_clusterConfig". But it makes very little difference.

getClusterConfig() is a different story. It is in set method, we should not call get method in a set method. It is just confusing.

Let's don't make it too complicated. My target is just to remove all the unnecessary private field references. And this specific method is a set method. It is necessary.

Reply:
I think I might need to clarify it a little bit more. Since you seem to be confused by the style things...

I'm not sure what is the "approach" that you refer to... Calling the method or referring to the private fields are both OK. They are actually the same thing, right? Unless the method has some concurrent control or so.

I don't see any reason why we have to use getClusterConfig() or _clusterConfig or clusterConfig in this context. They make a minor difference. If I have to compare getClusterConfig() is the worst, since it triggers one more method call.

The difference is whether you pass the **INPUT** as the method parameters or directly refer to the private fields. This makes a difference. The latter introduces hidden dependencies between different private things. So this style is not good.
This is the small improvement that I'm trying to do here.

And obviously, this rule is not applicable in the set methods. Because they have to refer to the private fields, right?

Reply:
Feel free to resolve this comment thread. The reason I ask is not for styling. It is that initially I not sure about the intention. I thought `getInstanceConfigMap()` had some tricks inside and that was why we did not use `_configMap`. If it is styling, no such intention, then either way is fine to me. Making them same style would also be good, but up to you.